File: javascript/jssearch/src/searchsettings.js
/*
 * searchsettings.js
 *
 * represents the settings to use when performing the search
 */

const {Color, FindSettings} = require('jsfind');

class SearchSettings extends FindSettings {
    firstMatch = false;
    inLinesAfterPatterns = [];
    inLinesBeforePatterns = [];
    lineColor = Color.GREEN;
    linesAfter = 0;
    linesAfterToPatterns = [];
    linesAfterUntilPatterns = [];
    linesBefore = 0;
    maxLineLength = 150;
    multilineSearch = false;
    outLinesAfterPatterns = [];
    outLinesBeforePatterns = [];
    printLines = false;
    printMatches = false;
    printResults = false;
    searchArchives = false;
    searchPatterns = [];
    textFileEncoding = "utf-8";
    uniqueLines = false;

    constructor() {
        super();
    }

    addInLinesAfterPatterns(pattern) {
        this.addPatterns(pattern, this.inLinesAfterPatterns);
    }

    addInLinesBeforePatterns(pattern) {
        this.addPatterns(pattern, this.inLinesBeforePatterns);
    }

    addLinesAfterToPatterns(pattern) {
        this.addPatterns(pattern, this.linesAfterToPatterns);
    }

    addLinesAfterUntilPatterns(pattern) {
        this.addPatterns(pattern, this.linesAfterUntilPatterns);
    }

    addOutLinesAfterPatterns(pattern) {
        this.addPatterns(pattern, this.outLinesAfterPatterns);
    }

    addOutLinesBeforePatterns(pattern) {
        this.addPatterns(pattern, this.outLinesBeforePatterns);
    }

    addSearchPatterns(pattern) {
        this.addPatterns(pattern, this.searchPatterns);
    }
}

exports.SearchSettings = SearchSettings;

File: ruby/rbsearch/lib/rbsearch/searchsettings.rb
require 'rbfind/color'
require 'rbfind/filetypes'
require 'rbfind/findsettings'

module RbSearch

  # SearchSettings - encapsulates search settings
  class SearchSettings < RbFind::FindSettings
    attr_accessor :first_match,
                  :in_lines_after_patterns,
                  :in_lines_before_patterns,
                  :line_color,
                  :lines_after,
                  :lines_after_to_patterns,
                  :lines_after_until_patterns,
                  :lines_before,
                  :max_line_length,
                  :multi_line_search,
                  :out_lines_after_patterns,
                  :out_lines_before_patterns,
                  :print_lines,
                  :print_matches,
                  :print_results,
                  :search_archives,
                  :search_patterns,
                  :text_file_encoding,
                  :unique_lines

    def initialize
      super
      @first_match = false
      @in_lines_after_patterns = Set.new
      @in_lines_before_patterns = Set.new
      @line_color = RbFind::Color::GREEN
      @lines_after = 0
      @lines_after_to_patterns = Set.new
      @lines_after_until_patterns = Set.new
      @lines_before = 0
      @max_line_length = 150
      @multi_line_search = false
      @out_lines_after_patterns = Set.new
      @out_lines_before_patterns = Set.new
      @print_lines = false
      @print_matches = false
      @print_results = true
      @search_archives = false
      @search_patterns = Set.new
      @text_file_encoding = 'utf-8'
      @unique_lines = false
    end

    def archives_only=(archives_only)
      @archives_only = archives_only
      @include_archives = archives_only if archives_only
      @search_archives = archives_only if archives_only
    end
  end
end
